File: week10_test/equi_filter.py
#!/usr/bin/python3

import sys,re
dic = {}
list = []
for line in sys.stdin.readlines():
	temp=[]
	a = re.split('\s+',line)
	for word in a:
		dic = {}
		flag = 0;
		for char in word:
			if char not in  dic:
				dic[char.lower()] = 1
			else:
				dic[char.lower()] += 1
				flag = 1
		count = {}
		for i in dic:
			count[dic[i]] = 1;
		if (not flag or len(count)) == 1:
			temp.append(word)
	str = ' '.join(temp)
	print(str)
	

File: week10_test/equi_filter.pl
#!/usr/bin/perl -w

use strict;

my @list = ();
my @temp =();
my %dic = ();
while(my $line = <STDIN>){
	chomp $line;
	@temp = ();
	my @a = split /\s+/, $line;
	foreach my $word (@a){
		%dic = ();
		my $flag = 0;
		my @b = split /\s*/, $word;
		foreach my $w (@b){
			$flag = 1 if(exists $dic{lc($w)});
			$dic{lc($w)}++;
		}
		my %count;   
        	foreach my $key (sort keys %dic){
                	$count{$dic{$key}} = 1;
	        }
		my $size = keys %count; 
		push @temp, "$word" if(!$flag || $size == 1);
	}
	@temp = join ' ', @temp; 
	push @temp, "\n";
	my $str = [@temp]; 
	push @list, $str;
}

foreach my $line (@list){
	print @$line;
}



